pkg/webhookconfig/registration.go

```diff
@@ -158,37 +158,35 @@ func (wrc *Register) Remove(cleanUp chan<- struct{}) {
 func (wrc *Register) UpdateWebhookConfigurations(configHandler config.Interface) {
 	logger := wrc.log.WithName("UpdateWebhookConfigurations")
 	for {
-		select {
-		case <-wrc.UpdateWebhookChan:
-			logger.Info("received the signal to update webhook configurations")
-
-			var nsSelector map[string]interface{}
-			webhookCfgs := configHandler.GetWebhooks()
-			if webhookCfgs != nil {
-				selector := webhookCfgs[0].NamespaceSelector
-				selectorBytes, err := json.Marshal(*selector)
-				if err != nil {
-					logger.Error(err, "failed to serialize namespaceSelector")
-					continue
-				}
-
-				if err = json.Unmarshal(selectorBytes, &nsSelector); err != nil {
-					logger.Error(err, "failed to convert namespaceSelector to the map")
-					continue
-				}
+		<-wrc.UpdateWebhookChan
+		logger.Info("received the signal to update webhook configurations")
+
+		var nsSelector map[string]interface{}
+		webhookCfgs := configHandler.GetWebhooks()
+		if webhookCfgs != nil {
+			selector := webhookCfgs[0].NamespaceSelector
+			selectorBytes, err := json.Marshal(*selector)
+			if err != nil {
+				logger.Error(err, "failed to serialize namespaceSelector")
```

Comment:
If the namespace selector is misconfigured, will be webhook still be created?

What is the expected syntax? 

Is there a way we can validate that earlier, like on startup?

Reply:
>If the namespace selector is misconfigured, will be webhook still be created?

Yes

>What is the expected syntax?

Currently it takes "a list of namespaceSelector in JSON format".

>Is there a way we can validate that earlier, like on startup?

Yes, we can parse the ConfigMap during the start. What should be the expected behavior? An error in the log or restarts of Kyverno?

Reply:
I added the validation logic, please refer to "Validate webhooks format" in the description above.